ProductsService/ProductsService.Application/UseCases/DiscountUseCases/Commands/AddDiscount/AddDiscountRequestHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using AutoMapper;
+using MediatR;
+using ProductsService.Application.Exceptions;
+using ProductsService.Application.Specifications.Discounts;
+using ProductsService.Domain.Abstractions.Database;
+using ProductsService.Domain.Abstractions.Specifications;
+using ProductsService.Domain.Entities;
+
+namespace ProductsService.Application.UseCases.DiscountUseCases.Commands.AddDiscount;
+
+internal class AddDiscountRequestHandler(IUnitOfWork unitOfWork, IMapper mapper)
+    : IRequestHandler<AddDiscountRequest>
+{
+    public async Task Handle(AddDiscountRequest request, CancellationToken cancellationToken)
+    {
+        var specification = new DiscountProductSpecification(request.ProductId);
+
+        var discount = await unitOfWork.DiscountQueryRepository.FirstOrDefaultAsync(specification, cancellationToken);
+
+        if (discount is not null)
+        {
+            throw new BadRequestException("This product already have discount");
+        }
+
+        discount = mapper.Map<Discount>(request);
```

Comment:
why do you need this mapping?

Reply:
I create this variable before this line to check, if discount for this product already exists. Line with mapping is accesable only if discount is null and I map a request into discount to add it into database